### DISCUSSION TITLE
Default export for route handlers in app router

### DISCUSSION THREAD

**Discussion by sadkebab** [vercel/next.js#56825] [Ideas]
### Goals

1. Give developers the ability to process all requests to a route with a single handler, instead of exporting a handler for every HTTP method.

### Background

Next.js is an amazing tool for frontend and full stack app... but it can be really usefull for building serverless backends very quickly, and with the vercel platform you can deploy them on edge trivially.

I was building an automatic crud library with drizzle-orm and so far in order to make it work the structure of the routes is:

`app/api/[table]/route.js (export GET, POST)`
`app/api/[table]/[id]/route.js (export GET, PUT, PATCH, DELETE)`

If I had the possibility to define a single default export and handle HTTP methods from there it would look something like this:

`app/api/[...crud]/route.js (export default)`

<img width="530" alt="Screenshot 2023-10-14 at 11 47 31" src="https://github.com/vercel/next.js/assets/44508455/85985c7f-a577-4547-a17b-1ebb24b941bf">

### Proposal

Next.js should check for handlers in named exports first, and then if it finds a default export it should apply it to the HTTP methods that were not in the named exports.

**Comment by reece727** [comment]
export { handler as POST, handler as GET, handler as PUT }
